<html>
<head>
<title>Custom annotations</title>
<meta name="order" content="6" />
</head>
<body>
<div>
  
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
  
  <h1 id="top"><span style="color:#006;padding-right:15px">Variant Effect Predictor</span><span style="color:#666"><img src="/i/16/page-user.png"> Custom annotations</span></h1>
  <hr/> 
        
  <p> VEP can integrate custom annotation from standard format
  files into your results by using the <a href="vep_options.html#opt_custom">--custom</a> flag. </p>
    
  <p> These files may be hosted locally or remotely, with no limit to the
  number or size of the files. The files must be indexed using the <a
  rel="external" href="http://samtools.sourceforge.net/tabix.shtml" target="_blank">tabix</a>
  utility (BED, GFF, GTF, VCF); bigWig files contain their own indices. </p>
  
  <p> Annotations typically appear as key=value pairs in the Extra column of the VEP
  output; they will also appear in the INFO column if using VCF format output.
  The value for a particular annotation is defined as the identifier for each
  feature; if not available, an identifier derived from the coordinates of the
  annotation is used. Annotations will appear in each line of output for the
  variant where multiple lines exist. </p>
  
  <h2 id="custom_formats">Data formats</h2>
  
  <p> VEP supports the following formats: </p>

  <ul>
    <li> Gene/transcript annotations - see <a href="vep_cache.html#gff">documentation</a>.<br/>
      <p style="margin:0.5em 0">Requires a <b>FASTA</b> file</p>
      <ul>
        <li><a rel="external" href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md">GFF</a> : a format for describing genes and other genomic features. </li>
        <li>
          <a href="/info/website/upload/gff.html" target="_blank">GTF</a> : a similar format derived from GFF.<br/>
        </li>
      </ul>
    </li>

    <li> Variant data
      <ul>
        <li><a
        rel="external" href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41"
        target="_blank">VCF</a> : a format used to describe genomic variants.
        VEP will use the 3rd column of the file as the identifier. INFO fields from records may be added to the VEP output. </li>
      </ul>
    </li>

    <li> Basic/uninterpreted data
      <ul>
        <li><a href="/info/website/upload/bed.html" target="_blank">BED</a> : a
        simple tab-delimited format containing 3-12 columns of data. The first 3
        columns contain the coordinates of the feature. If available, VEP
        will use the 4th column of the file as the identifier of the feature.
        </li>
        <li><a rel="external" href="http://genome.ucsc.edu/goldenPath/help/bigWig.html"
        target="_blank">bigWig</a> : a format for storage of dense continuous
        data. VEP uses the value for the given position as the "identifier".
        Note that bigWig files contain their own indices, and do not need to be
        indexed by tabix. Requires <a href="vep_download.html#bigfile">Bio::DB::BigFile</a>.</li>
      </ul>
    </li>
  </ul>
  
  <p> Any other files can be easily converted to be compatible with VEP;
  the easiest format to produce is a BED-like file containing coordinates and
  an (optional) identifier: </p>
  
  <pre class="code sh_sh">chr1    10000    11000    Feature1
chr3    25000    26000    Feature2
chrX    99000    99001    Feature3</pre>
  
  <p> Chromosomes can be denoted by either e.g. "chr7" or "7", "chrX" or "X". </p>
  
    <hr/>
  <h2 id="custom_prep">Preparing files</h2>
  
  <p> Custom annotation files must be prepared in a particular way in order to
  work with tabix and therefore with VEP. Files must be stripped of comment lines,
  sorted in chromosome and position order, compressed using bgzip
  and finally indexed using tabix. Here are some examples of that process for:</p>
  
  <ul>
    <li><b>GFF file</b>
  <pre class="code sh_sh">grep -v "#" myData.gff | sort -k1,1 -k4,4n -k5,5n -t$'\t' | bgzip -c > myData.gff.gz
tabix -p gff myData.gff.gz</pre></li>
    <li><b>BED file</b>
  <pre class="code sh_sh">grep -v "#" myData.bed | sort -k1,1 -k2,2n -k3,3n -t$'\t' | bgzip -c > myData.bed.gz
tabix -p bed myData.bed.gz</pre>
  </ul>  

  <p> The tabix utility has several preset filetypes that it can process, and
  it can also process any arbitrary filetype containing at least a chromosome
  and position column. See the <a
  rel="external" href="http://samtools.sourceforge.net/tabix.shtml"
  target="_blank">documentation</a> for details. </p>
  
  <p> If you are going to use the file remotely (i.e. over HTTP or FTP
  protocol), you should ensure the file is world-readable on your server. </p>
  
  <hr/>
  <h2 id="custom_options">Options</h2>
  
  <p> Each custom file that you configure VEP to use can be configured.
  Beyond the filepath, there are further options, each of which is specified
  in a comma-separated list, like this: </p>
  
  <pre class="code sh_sh">
./vep [...] --custom <b>Filename</b> , <b>Short_name</b> , <b>File_type</b> , <b>Annotation_type</b> , <b>Force_report_coordinates</b> , <b>VCF_fields</b></pre>
  
  <p> The options are as follows: </p>
  
  <ul>
    <li><b>Filename</b> :
    <div style="margin:5px 0px 12px">The path to the file. For tabix indexed files, the
    VEP will check that both the file and the corresponding .tbi file exist.
    For remote files, VEP will check that the tabix index is accessible on startup.</div></li>

    <li><b>Short name</b> : 
    <div style="margin:5px 0px 12px">A name for the annotation that will appear as
    the key in the key=value pairs in the results.<br />If not defined, this will 
    default to the annotation filename for the first set of annotation added (e.g. "myPhenotypes.bed.gz" in the second example below if the short name was missing).</div></li>

    <li><b>File type</b> :
      <div style="margin:5px 0px 12px">
        <div style="float:left">
          <pre class="code sh_sh" style="margin:4px 0px">"bed", "gff", "gtf", "vcf" or "bigwig"</pre>
        </div>
        <div style="float:left;padding:8px 10px;margin:4px 0px">
          (if not specified, VEP assumes the file is in <b>BED</b> format)
        </div>
        <div class="clear"></div>
      </div>
    </li>

    <li><b>Annotation type</b> :
      <div style="margin:5px 0px 12px">
        <div>
          <div style="float:left">
            <pre class="code sh_sh" style="margin:4px 0px">"exact" or "overlap"</pre>
          </div>
          <div style="float:left;padding:8px 10px;margin:4px 0px">
            (if left blank, assumed to be <b>overlap</b>) 
          </div>
          <div class="clear"></div>
        </div>
        <div>
          When using "<span style="color:blue">exact</span>" only annotations whose coordinates match exactly those of the
          variant will be reported. This would be suitable for position specific
          information such as conservation scores, allele frequencies or phenotype
          information. Using "<span style="color:blue">overlap</span>", any annotation that overlaps the variant
          by even 1bp will be reported.
        </div>
      </div>
    </li>
    
    <li><b>Force report coordinates</b> :
      <div style="margin:5px 0px 12px">
        <div>
          <div style="float:left">
            <pre class="code sh_sh" style="margin:4px 0px">"0" or "1"</pre>
          </div>
          <div style="float:left;padding:8px 10px;margin:4px 0px">
            (if left blank, assumed to be <b>0</b>)
          </div>
          <div class="clear"></div>
        </div>
        <div>If set to "1", this forces VEP to output the
          coordinates of an overlapping custom feature instead of any found
          identifier (or value in the case of bigWig) field. If set to "0" (the
          default), VEP will output the identifier field if one is found; if
          none is found, then the coordinates are used instead.
        </div>
      </div>
    </li>
      
    <li><b>VCF fields</b> :
      <div style="margin:5px 0px 12px">You can specify any info type (e.g. "AC") present in the INFO field of the custom input VCF, to add these as custom annotations:
        <ul>
          <li>If using "<span style="color:blue">exact</span>" annotation type, allele-specific annotation will be retrieved.</li>
          <li>The INFO field name will be prefixed with the short name, e.g.
              using short name "test", the INFO field "foo" will appear as "test_FOO" in the VEP output.</li>
          <li>In VCF files the custom annotations are added to the CSQ INFO field.</li>
          <li>Alleles in the input and VCF entry are trimmed in both directions in an attempt to match complex or poorly formatted entries.</li>
        </ul>
      </div>
    </li>
  </ul>

  <p>For example: </p>
  
  <pre class="code sh_sh">
# BigWig file
./vep [...] --custom frequencies.bw,Frequency,bigwig,exact,0
# BED file
./vep [...] --custom http://www.myserver.com/data/myPhenotypes.bed.gz,Phenotype,bed,exact,1
# VCF file
./vep [...] --custom ftp://ftp.ensembl.org/pub/data_files/homo_sapiens/GRCh37/variation_genotype/TOPMED_GRCh37.vcf.gz,,vcf,exact,0,TOPMED</pre>
  <br />

  <hr/>
  <h2 id="custom_remote">Using remote files</h2>
  
  <p>The tabix utility makes it possible to read annotation files from remote
  locations, for example over HTTP or FTP protocols.</p>

  <p>In order to do this, the
  .tbi index file is downloaded locally (to the current working directory)
  when VEP is run. From this point on, only the portions of data requested
  by VEP (i.e. those overlapping the variants in your input file) are
  downloaded.</p>
 
  <div style="width:auto" class="warning">

    <h3>Be aware</h3>

    <div class="message-pad">
      <p>1. It is still possible to cause problems with network traffic in this manner by requesting data for a
         large number of variants.</p>
      
      <p>2. Users with large amounts of data should download the annotation file locally rather than risk causing any issues!</p>
    </div>
  </div>  
 
  <p> bigWig files can also be used remotely in the same way as tabix-indexed
  files, although less stringent checks are carried out on VEP startup. </p>
  
</div>

</body>
</html>
