<p>The orange clownfish (Amphiprion percula) also known as percula clownfish and clown anemonefish, is widely known as a popular aquarium fish. Like other clownfishes (also known as anemonefishes), it often lives in association with sea anemones. A. percula is associated specifically with Heteractis magnifica and Stichodactyla gigantea, and as larvae use chemical cues released from the anemones to identify and locate the appropriate host species to use them for shelter and protection. This causes preferential selection when finding their anemone host species. Although popular, maintaining this species in captivity is rather complex. The Great Barrier Reef Marine Park Authority regulates the number of collection permits issued to aquarium fish dealers who seek this, and other tropical fish within the Great Barrier Reef Marine Park.
The symbiosis between anemonefish and anemones depends on the presence of the fish drawing other fish to the anemone, where they are stung by its venomous tentacles. The anemone helps the fish by giving it protection from predators, which include brittle stars, wrasses, and other damselfish, and the fish helps the anemone by feeding it, increasing oxygenation, and removing waste material from the host. Various hypotheses exist about the fish&#39;s ability to live within the anemone without being harmed. One study carried out at Marineland of the Pacific by Dr. Demorest Davenport and Dr. Kenneth Noris in 1958 revealed that the mucus secreted by the anemone fish prevented the anemone from discharging its lethal stinging nematocysts. A second hypothesis is that A. percula has acquired immunity towards the sea anemone&rsquo;s toxins, and  a combination of the two has been shown to be the case. The fish feed on algae, zooplankton, worms, and small crustaceans.</p>
<p>The gene annotation process was carried out using a combination of protein-to-genome alignments, annotation mapping from a suitable reference species and RNA-seq alignments (where RNA-seq data with appropriate meta data were publicly available). For each candidate gene region, a selection process was applied to choose the most appropriate set of transcripts based on evolutionary distance, experimental evidence for the source data and quality of the alignments.</br>Small ncRNAs were obtained using a combination of BLAST and Infernal/RNAfold.</br> Pseudogenes were calculated by looking at genes with a large percentage of non-biological introns (introns of &lt;10bp), where the gene was covered in repeats, or where the gene was single exon and evidence of a functional multi-exon paralog was found elsewhere in the genome.</br> lincRNAs were generated via RNA-seq data where no evidence of protein homology or protein domains could be found in the transcript.</p><p>In accordance with the <a href="https://en.wikipedia.org/wiki/Fort_Lauderdale_Agreement">Fort Lauderdale Agreement </a>, please check the publication status of the genome/assembly before publishing any genome-wide analyses using these data.</p>
