<p>The gene annotation process was carried out using a combination of alignments to UniProt proteins and RNA-seq alignments (to RNASeq data from 39 tissues). 

For each candidate gene region, a selection process was applied to choose the most appropriate set of transcripts based on evolutionary distance, experimental evidence for the source data and quality of the alignments. 
Small ncRNAs were obtained using a combination of BLAST and Infernal/RNAfold. 

Pseudogenes were calculated by looking at genes with a large percentage of non-biological introns (introns of &lt;10bp), where the gene was covered in repeats, or where the gene was single exon and evidence of a functional multi-exon paralog was found elsewhere in the genome.</p>

<ul>
  <li><a href="/info/genome/genebuild/2018_03_goat_genebuild.pdf">Detailed information on the goat genebuild</a> (PDF)</li>
</ul>

<p>In accordance with the <a href="https://en.wikipedia.org/wiki/Fort_Lauderdale_Agreement">Fort Lauderdale Agreement </a>, please check the publication status of the genome/assembly before publishing any genome-wide analyses using these data.</p>
