<p>The desert tortoises (Gopherus agassizii and Gopherus morafkai) are two species of tortoise native to the Mojave and Sonoran Deserts of the southwestern United States and northwestern Mexico and the Sinaloan thornscrub of northwestern Mexico. G. agassizii is distributed in western Arizona, southeastern California, southern Nevada, and southwestern Utah.  The specific name agassizii is in honor of Swiss-American zoologist Jean Louis Rodolphe Agassiz. Recently, on the basis of DNA, geographic, and behavioral differences between desert tortoises east and west of the Colorado River, it was decided that two species of desert tortoises exist: Agassiz&#39;s desert tortoise (Gopherus agassizii ) and Morafka&#39;s desert tortoise (Gopherus morafkai). G. morafkai occurs east of the Colorado River in Arizona, as well as in the states of Sonora and Sinaloa, Mexico. This species may be a composite of two species.The new species name is in honor of the late Professor David Joseph Morafka of California State University, Dominguez Hills, in recognition of his many contributions to the study and conservation of Gopherus.The desert tortoise lives about 50 to 80 years; it grows slowly and generally has a low reproductive rate.  It spends most of its time in burrows, rock shelters, and pallets to regulate body temperature and reduce water loss.  It is most active after seasonal rains and is inactive during most of the year. This inactivity helps reduce water loss during hot periods, whereas winter hibernation facilitates survival during freezing temperatures and low food availability. Desert tortoises can tolerate water, salt, and energy imbalances on a daily basis, which increases their lifespans.</p>
<p>The gene annotation process was carried out using a combination of protein-to-genome alignments, annotation mapping from a suitable reference species and RNA-seq alignments (where RNA-seq data with appropriate meta data were publicly available). For each candidate gene region, a selection process was applied to choose the most appropriate set of transcripts based on evolutionary distance, experimental evidence for the source data and quality of the alignments.</br>Small ncRNAs were obtained using a combination of BLAST and Infernal/RNAfold.</br> Pseudogenes were calculated by looking at genes with a large percentage of non-biological introns (introns of &lt;10bp), where the gene was covered in repeats, or where the gene was single exon and evidence of a functional multi-exon paralog was found elsewhere in the genome.</br> lincRNAs were generated via RNA-seq data where no evidence of protein homology or protein domains could be found in the transcript.</p><p>In accordance with the <a href="https://en.wikipedia.org/wiki/Fort_Lauderdale_Agreement">Fort Lauderdale Agreement</a>, please check the publication status of the genome/assembly before publishing any genome-wide analyses using these data.</p>
