<p>The tiger tail seahorse (Hippocampus comes) is a species of fish in the family Syngnathidae. It is found in India,
  Indonesia, Malaysia, the Philippines, Singapore, Thailand and Vietnam. Its natural habitats are subtidal aquatic beds
  and coral reefs. It is threatened by habitat loss.The tiger tail sea horse lives in Western Central Pacific:
  Malaysia, Singapore, Thailand, Vietnam, and the Philippines. It lives from 0-1.5 years in captivity and in the wild,
  1&ndash;5 years. It is harmless. Its climate in water is tropical; 15&deg;N - 1&deg;N and Its maximum size is
  18.7&nbsp;cm. Its snout is 2.2 in head length; it is used to suck up food. They eat small fish, coral, small shrimp,
  and plankton. The most common pattern is alternating yellow and black. The tail has stripes from the belly to the tip
  of the tail.
  These sea horses are normally found in pairs on coral reefs, sponge gardens, kelp, or floating Sargassum, this
  species is nocturnal. The male carries the eggs in a brood pouch on their chest which holds from 1 - 2,000 eggs and
  the pregnancy takes from 1 to 4 weeks.
  It is also used for traditional Chinese medicine. Seahorse populations are thought to have been endangered in recent
  years by over fishing and habitat destruction. The seahorse is used in traditional Chinese medicine, and as many as
  20 million seahorses may be caught each year and sold for this purpose. Import and export of seahorses has been
  controlled under CITES since May 15, 2004.
  They don&#39;t have scales as fish do, they have a tough thin skin stretched out around bony rings on their bodies.
  This makes them an unlikely prey for most marine animals, as they are too bony to digest. They swim upright, rather
  than horizontally.</p>
<p>The gene annotation process was carried out using a combination of protein-to-genome alignments, annotation mapping
  from a suitable reference species and RNA-seq alignments (where RNA-seq data with appropriate meta data were publicly
  available). For each candidate gene region, a selection process was applied to choose the most appropriate set of
  transcripts based on evolutionary distance, experimental evidence for the source data and quality of the alignments.
  Small ncRNAs were obtained using a combination of BLAST and Infernal/RNAfold. Pseudogenes were calculated by looking
  at genes with a large percentage of non-biological introns (introns of &lt;10bp), where the gene was covered in
  repeats, or where the gene was single exon and evidence of a functional multi-exon paralog was found elsewhere in the
  genome. lincRNAs were generated via RNA-seq data where no evidence of protein homology or protein domains could
  be found in the transcript.</p>

<ul>
  <li>
    <a href="/info/genome/genebuild/2018_12_fish_clade_gene_annotation.pdf">Detailed information on fish clade
      genebuild</a>
    (PDF)
  </li>
</ul>

<p>
  In accordance with the
  <a href="https://en.wikipedia.org/wiki/Fort_Lauderdale_Agreement">Fort Lauderdale Agreement</a>, please check the
  publication status of the genome/assembly before publishing any genome-wide analyses using these data.
</p>