<p>The tammar wallaby (Macropus eugenii), also known as the dama wallaby or darma wallaby, is a small macropod native to South and Western Australia. Though its geographical range has been severely reduced since European colonisation, the tammar remains common within its reduced range and is listed as &quot;Least Concern&quot; by the International Union for Conservation of Nature (IUCN). It has been introduced to New Zealand and reintroduced to some areas of Australia where it had been previously eradicated. Skull differences distinguish tammars from Western Australia, Kangaroo Island and mainland South Australia, making them distinct population groups or possibly different subspecies.The tammar is among the smallest of the wallabies in the genus Macropus. Its coat colour is largely grey. The tammar has several notable adaptations, including the ability to retain energy while hopping, colour vision and the ability to drink seawater. A nocturnal species, it spends nighttime in grassland habitat and daytime in shrubland. It is also very gregarious and has a seasonal, promiscuous mating pattern. A female tammar can nurse a joey in her pouch while keeping an embryo in her uterus. The tammar is a model species for research on marsupials, and on mammals in general. It is one of many organisms to have had its genome sequenced.</p>
<p>The gene annotation process was carried out using a combination of protein-to-genome alignments, annotation mapping from a suitable reference species and RNA-seq alignments (where RNA-seq data with appropriate meta data were publicly available). For each candidate gene region, a selection process was applied to choose the most appropriate set of transcripts based on evolutionary distance, experimental evidence for the source data and quality of the alignments.</br>Small ncRNAs were obtained using a combination of BLAST and Infernal/RNAfold.</br> Pseudogenes were calculated by looking at genes with a large percentage of non-biological introns (introns of &lt;10bp), where the gene was covered in repeats, or where the gene was single exon and evidence of a functional multi-exon paralog was found elsewhere in the genome.</br> lincRNAs were generated via RNA-seq data where no evidence of protein homology or protein domains could be found in the transcript.</p><p>In accordance with the <a href="https://en.wikipedia.org/wiki/Fort_Lauderdale_Agreement">Fort Lauderdale Agreement </a>, please check the publication status of the genome/assembly before publishing any genome-wide analyses using these data.</p>
