<p>
Sscrofa11.1 was annotated using the standard Ensembl automatic gene
annotation system, incorporating RNA-Seq data and PacBio long reads
provided by the SGSC.
The gene annotation process was carried out using a combination of
cDNA alignments, protein-to-genome alignments, PacBio long reads and
RNA-seq alignments. For each candidate gene region, a selection process
was applied to choose the most appropriate set of transcripts based on
evolutionary distance, experimental evidence for the source data and
quality of the alignments.
</br>
Small ncRNAs were obtained using a combination of BLAST and Infernal/RNAfold.
Pseudogenes were calculated by looking at genes with a large percentage
of non-biological introns (introns of &lt;10bp), where the gene was covered in
repeats, or where the gene was single exon and evidence of a functional
multi-exon paralog was found elsewhere in the genome. lincRNAs were generated
via transcriptomic data where no evidence of protein homology or protein
domains could be found in the transcript.
</p>

<p>
In accordance with the <a href="https://en.wikipedia.org/wiki/Fort_Lauderdale_Agreement" rel="external">Fort Lauderdale Agreement</a>,
please check the publication status of the genome/assembly before publishing any genome-wide analyses using these data.
</p>

<ul>
  <li><a href="/info/genome/genebuild/2019_09_sus_scrofa_gene_annotation.pdf">Detailed information on pig genebuild</a> (PDF)</li>                                                                                                                               
</ul>

<h3>PacBio long read data set</h3>

<p>Nine tissue samples were sequenced using the PacBio long read sequencing
technology, brain, diaphragm, hypothalamus, liver, longissimus muscle, pituitary,
small intestine, spleen, thymus. All sets were sequenced using 3' capping and
were used to add UTR to gene models and as input source for our lincRNA discovery
pipeline.
</br>
Furthermore we collapsed the models created to produce a non-redundant set. We did
a BLASTp of these models against UniProt proteins of vertabrate species with protein
existence level 1 and 2 in order to confirm the open reading frame (ORF). We also
checked the splice junctions with the RNA-seq data set. The models with a high
coding potential and with full RNA-seq support were used as input for the gene model
generation.</p>

<p>The tissue-specific sets of aligned sequences are as follows:</p>

<table class="ss tint">
  <tr><th>Tissue</th><th>Number of aligned sequences</th></tr>
  <tr class="bg1"><td>Brain</td><td>354494</td></tr>
  <tr class="bg2"><td>Diaphragm</td><td>391813</td></tr>
  <tr class="bg1"><td>Hypothalamus</td><td>256930</td></tr>
  <tr class="bg2"><td>Liver</td><td>491796</td></tr>
  <tr class="bg1"><td>Longissimus muscle</td><td>361864</td></tr>
  <tr class="bg2"><td>Pituitary</td><td>252707</td></tr>
  <tr class="bg1"><td>Small intestine</td><td>406144</td></tr>
  <tr class="bg2"><td>Spleen</td><td>449425</td></tr>
  <tr class="bg1"><td>Thymus</td><td>374515</td></tr>
</table>

<h3>RNASeq data set</h3>

<p>In addition to the main set, we have predicted gene models for each
tissue type using the RNA-Seq pipeline. We did a BLASTp of these models
against UniProt proteins of vertabrate species with protein existence
level 1 and 2 in order to confirm the open reading frame (ORF). The best
BLAST hit is displayed as a transcript supporting evidence. The data was
also used to add UTR to gene models.</p>

<p>The tissue-specific sets of transcript models built using our RNAseq
pipeline are as follows:</p>

<table class="ss tint">
  <tr><th>Tissue</th><th>Number of gene models</th></tr>
  <tr class="bg1"><td>Alveolar macrophages</td><td>34867</td></tr>
  <tr class="bg2"><td>Amygdala</td><td>38118</td></tr>
  <tr class="bg1"><td>Brain stem</td><td>35791</td></tr>
  <tr class="bg2"><td>Caecum</td><td>40716</td></tr>
  <tr class="bg1"><td>Cerebellum</td><td>36132</td></tr>
  <tr class="bg2"><td>Colon</td><td>34520</td></tr>
  <tr class="bg1"><td>Corpus callosum</td><td>37474</td></tr>
  <tr class="bg2"><td>Duodenum</td><td>40112</td></tr>
  <tr class="bg1"><td>Epididymis</td><td>37377</td></tr>
  <tr class="bg2"><td>Frontal lobe</td><td>35119</td></tr>
  <tr class="bg1"><td>Hippocampus</td><td>36403</td></tr>
  <tr class="bg2"><td>Ileum</td><td>36661</td></tr>
  <tr class="bg1"><td>Kidney cortex</td><td>35544</td></tr>
  <tr class="bg2"><td>Left ventricle</td><td>33125</td></tr>
  <tr class="bg1"><td>Mesenteric lymph node</td><td>40250</td></tr>
  <tr class="bg2"><td>Medulla oblongata</td><td>42716</td></tr>
  <tr class="bg1"><td>Merged</td><td>54258</td></tr>
  <tr class="bg2"><td>Occipital lobe</td><td>35390</td></tr>
  <tr class="bg1"><td>Omentum</td><td>27570</td></tr>
  <tr class="bg2"><td>Penis</td><td>37121</td></tr>
  <tr class="bg1"><td>Pituitary gland</td><td>33800</td></tr>
  <tr class="bg2"><td>Pituitary</td><td>35482</td></tr>
  <tr class="bg1"><td>Pons</td><td>35974</td></tr>
  <tr class="bg2"><td>Skeletal muscle</td><td>32011</td></tr>
  <tr class="bg1"><td>Spleen</td><td>35130</td></tr>
  <tr class="bg2"><td>Stomach</td><td>33326</td></tr>
  <tr class="bg1"><td>Thalamus</td><td>36047</td></tr>
  <tr class="bg2"><td>Tonsil</td><td>38154</td></tr>
  <tr class="bg1"><td>Uterus</td><td>31178</td></tr>
</table>
