<p>The polar bear (Ursus maritimus) is a hypercarnivorous bear whose native range lies largely within the Arctic
  Circle, encompassing the Arctic Ocean, its surrounding seas and surrounding land masses. It is a large bear,
  approximately the same size as the omnivorous Kodiak bear (Ursus arctos middendorffi). A boar (adult male) weighs
  around 350&ndash;700&nbsp;kg (772&ndash;1,543&nbsp;lb), while a sow (adult female) is about half that size. Although
  it is the sister species of the brown bear, it has evolved to occupy a narrower ecological niche, with many body
  characteristics adapted for cold temperatures, for moving across snow, ice and open water, and for hunting seals,
  which make up most of its diet. Although most polar bears are born on land, they spend most of their time on the sea
  ice. Their scientific name means &quot;maritime bear&quot; and derives from this fact. Polar bears hunt their
  preferred food of seals from the edge of sea ice, often living off fat reserves when no sea ice is present. Because
  of their dependence on the sea ice, polar bears are classified as marine mammals.Because of expected habitat loss
  caused by climate change, the polar bear is classified as a vulnerable species, and at least three of the nineteen
  polar bear subpopulations are currently in decline. However, at least two of the nineteen subpopulations are
  currently increasing, while another six are considered stable. For decades, large-scale hunting raised international
  concern for the future of the species, but populations rebounded after controls and quotas began to take effect. For
  thousands of years, the polar bear has been a key figure in the material, spiritual, and cultural life of circumpolar
  peoples, and polar bears remain important in their cultures. Historically, the polar bear has also been known as the
  white bear.</p>
<p>The gene annotation process was carried out using a combination of protein-to-genome alignments, annotation mapping
  from a suitable reference species and RNA-seq alignments (where RNA-seq data with appropriate meta data were publicly
  available). For each candidate gene region, a selection process was applied to choose the most appropriate set of
  transcripts based on evolutionary distance, experimental evidence for the source data and quality of the alignments.
  Small ncRNAs were obtained using a combination of BLAST and Infernal/RNAfold. Pseudogenes were calculated by looking
  at genes with a large percentage of non-biological introns (introns of &lt;10bp), where the gene was covered in
  repeats, or where the gene was single exon and evidence of a functional multi-exon paralog was found elsewhere in the
  genome. lincRNAs were generated via RNA-seq data where no evidence of protein homology or protein domains could
  be found in the transcript.</p>

<ul>
  <li>
    <a href="/info/genome/genebuild/2018_12_mammal_clade_gene_annotation.pdf">Detailed information on mammal clade
      genebuild</a>
    (PDF)
  </li>
</ul>

<p>
  In accordance with the
  <a href="https://en.wikipedia.org/wiki/Fort_Lauderdale_Agreement">Fort Lauderdale Agreement</a>, please check the
  publication status of the genome/assembly before publishing any genome-wide analyses using these data.
</p>